### PR TITLE
Reduce the amount of errors logged, primarily in Node.js/Travis, when running the unit-tests

### PR DESCRIPTION
**Pull request by Snuffleupagus**
*Split off from PR #9729, as requested in https://github.com/mozilla/pdf.js/pull/9729#issuecomment-393698402.* /cc @timvandermeij 

Compare the output without these patches https://travis-ci.org/mozilla/pdf.js/builds/386523654, to the output *with* the patches https://travis-ci.org/mozilla/pdf.js/builds/386612114.

### CODE REVIEW

File: src/display/api_compatibility.js [hunk 1]
```diff
@@ -42,9 +44,15 @@ if (typeof PDFJSDev === 'undefined' || PDFJSDev.test('GENERIC')) {
       compatibilityParams.disableStream = true;
     }
   })();
+
+  // Support: Node.js
+  (function checkFontFaceAndImage() {
+    // Node.js is missing native support for `@font-face` and `Image`.
+    if (isNodeJS()) {
+      compatibilityParams.disableFontFace = true;
```
Comment (timvandermeij) [inline src/display/api_compatibility.js]: Now that this is set in the library, I think we can simplify:

https://github.com/mozilla/pdf.js/blob/57e8676180752c375e0d364234565d6240a9a3ec/examples/node/pdf2png/pdf2png.js#L62-L66

to:

```javascript
pdfjsLib.getDocument({ data: rawData, }).then(function (pdfDocument) {
```

**Comment by timvandermeij** [comment]
Aside from the comment above, this looks good to me. I'll try to review this and the other PR tomorrow.

**Review by timvandermeij — COMMENTED** [review]
I finished reviewing this. If the example comment is adressed, this is good to go.

File: src/display/api_compatibility.js (hunk 1 above)
Comment (Snuffleupagus) [reply src/display/api_compatibility.js]: > [...] so please check if the compatibility introduced here is indeed included in the library bundle.

Given that https://github.com/mozilla/pdf.js/blob/master/src/display/api_compatibility.js is imported by the API, this should be totally fine; see also https://github.com/mozilla/pdf.js/blob/2921cc0eba053896860e62a4bd1ffb5d29be51e5/src/display/api.js#L29

Furthermore, that fact that the `document` errors are now gone from the Travis output also shows that including these new Node.js specific compatibility options works fine.

**Comment by Snuffleupagus** [comment]
There's apparently still some warnings being printed in the Node.js/Travis output, coming from unit-testing `web/` files (since only the `src/` files use `info`/`warn`).
While the viewer shouldn't be using those helper functions, it probably wouldn't be *that* difficult to add (new) viewer-specific helper functions such that the viewer files wouldn't cause console "spam" either.

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Nice work, thank you!

**Pull request merged**